ui/lib/utils.ts

```diff
@@ -108,6 +108,13 @@ export function formatMetadataKey(key: string): string {
 
 export const convertImage = (image: string) => {
```

Comment:
This function is somewhat hard to read. I suggest to refactor top-level else-ifs into just top-level ifs with returns. For example, instead of:

```
if (prefix === "ghcr.io" || prefix === "gcr.io") return "https://" + noTag;
else if (prefix === "quay.io") {
  url = "https://quay.io/repository/";
  return makeUrl(url, split);
}
```

we could have

```
if (prefix === "ghcr.io" || prefix === "gcr.io") return "https://" + noTag;
if (prefix === "quay.io") {
  url = "https://quay.io/repository/";
  return makeUrl(url, split);
}
```

This would improve readability.

Reply:
And probably some of the nested if-elses could be simplified like that too (replaces with nested ifs).

Reply:
brb refactoring